main.tf

```diff
@@ -0,0 +1,102 @@
+locals {
+  hostname = format("%s-bastion", var.bastion_name)
+}
+
+// Dedicated service account for the Bastion instance.
+resource "google_service_account" "bastion" {
+  account_id   = format("%s-bastion-sa", var.bastion_name)
+  display_name = "GKE Bastion Service Account"
+}
+
+// Allow access to the Bastion Host via SSH.
+resource "google_compute_firewall" "bastion-ssh" {
+  name          = format("%s-bastion-ssh", var.bastion_name)
+  network       = var.network_name
+  direction     = "INGRESS"
+  project       = var.project_id
+  source_ranges = ["0.0.0.0/0"] // TODO: Restrict further.
+
+  allow {
+    protocol = "tcp"
+    ports    = ["22"]
+  }
+
+  target_tags = ["bastion"]
+}
+
+// The user-data script on Bastion instance provisioning.
+data "template_file" "startup_script" {
+  template = <<-EOF
+  sudo apt-get update -y
+  sudo apt-get install -y tinyproxy
+  EOF
+}
+
+// The Bastion host.
+resource "google_compute_instance" "bastion" {
+  name         = local.hostname
+  machine_type = "e2-micro"
+  zone         = var.zone
+  project      = var.project_id
+  tags         = ["bastion"]
+
+  boot_disk {
+    initialize_params {
+      image = "debian-cloud/debian-10"
```

Comment:
image should be a variable. So user can spawn bastion as per there own choice of OS